magen_user/tests/user_test.py

```diff
@@ -0,0 +1,188 @@
+# coding=utf-8
+"""User test for password checks"""
+
+import http
+import unittest
+from unittest import mock
+
+from magen_user_api import config
+from ..magen_user_api import db, user_api
+from ..magen_user_api.user_model import UserModel
+
+
+class TestUser(unittest.TestCase):
+
+    def setUp(self):
+        config.app.config['TESTING'] = True
+        config.app.config['WTF_CSRF_ENABLED'] = False
+        config.app.config['SECRET_KEY'] = 'test_key'
+        config.app.config['SECURITY_PASSWORD_SALT'] = 'test_salt'
+        config.app.config['SERVER_NAME'] = 'app'
+        config.app.register_blueprint(user_api.users_bp)
+        config.app.register_blueprint(user_api.main_bp)
+
+        self.test_app = config.app.test_client()
+        with db.connect(config.TEST_DB_NAME) as db_instance:
+            db_instance.drop_collection(config.USER_COLLECTION_NAME)
+        config.DEV_DB_NAME = config.TEST_DB_NAME
+
+    def tearDown(self):
+        with db.connect(config.TEST_DB_NAME) as db_instance:
+            db_instance.drop_collection(config.USER_COLLECTION_NAME)
+
+    def test_register_EqualPasswords(self):
+        data = {'email': 'test@test.com', 'password': 'testtest1', 'confirm': 'testtest1'}
+        # Register user
+        with mock.patch('magen_user.magen_user_api.user_api.send_confirmation'):
+            self.test_app.post(
+                '/register/',
+                data=data
+            )
+        # Same password and confirm password
+        self.assertIs(data['password'], data['confirm'])
+
+        # Checking password hashing
+        with db.connect(config.DEV_DB_NAME) as db_instance:
+            result = UserModel.select_by_email(db_instance, data['email'])
+            self.assertEqual(result.count, 1)
+            user = result.documents
+            self.assertTrue(user_api.check_password_hash(user.password, user.salt, data['password'].encode('utf-8')))
+
+    def test_register_NotEqualPasswords(self):
```

Comment:
This test does not exercise anything because the mock.patch is irrelevant. We are asserting on two variables  the test itself created.